clover_admin.rb

```diff
@@ -182,22 +182,104 @@ def self.object_action(...)
     form_options(wrapper: :div)
 
     link = lambda do |obj|
+      return "" unless obj
       "<a href=\"/model/#{obj.class}/#{obj.ubid}\">#{Erubi.h(obj.name)}</a>"
     end
 
     show_html do |obj, column|
       case column
       when :name
         link.call(obj)
-      when :project, :location
+      when :project, :location, :vm_host
         link.call(obj.send(column))
       end
     end
 
+    model Account do
+      order Sequel.desc(Sequel[:accounts][:created_at])
+      eager [:identities]
+      columns [:name, :email, :status_id, :provider_names, :created_at, :suspended_at]
+      column_options email: {type: "text"},
+        status_id: {type: "select", options: {Unverified: 1, Verified: 2, Closed: 3}, add_blank: true},
+        provider_names: {label: "Providers", type: "select", options: ["google", "github"], add_blank: true},
+        created_at: {type: "text"},
+        suspended_at: {label: "Suspended", type: "boolean", value: nil}
+
+      column_search_filter do |ds, column, value|
+        if column == :provider_names
+          ds.join(:account_identities, account_id: :id).where(provider: value)
```

Comment:
Assuming you switch to `eager_graph`:

```suggestion
          ds.where(provider: value)
```